most-used-words.el

```diff
@@ -50,97 +50,170 @@
 
 (require 'cl-lib)
 
+(defgroup most-used-words nil
+  "Display most used words in buffer."
+  :prefix "most-used-words-"
+  :group 'tool
+  :link '(url-link :tag "Repository" "https://github.com/udyantw/most-used-words"))
+
+(defcustom most-used-words-word-display 3
+  "Default word data to display."
+  :type 'integer
+  :group 'most-used-words)
+
+(defcustom most-used-words-display-type 'buffer
+  "The method to display most used words."
+  :type '(choice (const :tag "buffer" buffer)
+                 (const :tag "table" table))
+  :group 'most-used-words)
+
+(defvar most-used-words--buffer-name "*Most used words*"
+  "Buffer name for displaying most used words.")
+
+(defvar most-used-words--buffer nil
+  "Recorded buffer for currently doing most used words.")
+
+;;; Util
+
 (cl-defmacro most-used-words-with-view-buffer (buffer &body body)
+  "Default BUFFER for this package to display and execute BODY."
   `(progn
-     (with-current-buffer ,buffer
-       ,@body)
-     (view-buffer-other-window ,buffer
-			       nil
-			       #'kill-buffer)))
+     (with-current-buffer ,buffer ,@body)
+     (view-buffer-other-window ,buffer nil #'kill-buffer)))
+
+(defun most-used-words--form-data (word count percent)
+  "Form the plist data item with WORD, COUNT, PERCENT."
+  (list :word word :count count :percent percent))
+
+(defun most-used-words-data (n)
+  "Form used word data base on N count of data displayed."
+  (let* ((most-used (most-used-words-buffer-1 n t))
+         (word-counts (cl-first most-used))
+         (total-count (float (cl-second most-used)))
+         (data-lst '())  data-item)
+    (cl-loop for (word count) in word-counts
+             do
+             (setq data-item (most-used-words--form-data word count (* 100 (/ count total-count))))
+             (push data-item data-lst))
+    (reverse data-lst)))
```

Comment:
This can be externally use, I think this is great to wrap it into a public function so that the other people who might interested to this package can use the data/information externally! 😃

Reply:
Looks good. One small point if it suits you: could you write the variable as 'data-list' instead of 'data-lst'?

Reply:
On second thoughts, please see if this stylistic rewrite is fine:

(defun most-used-words-data (n)
  "Form used word data base on N count of data displayed."
  (let* ((most-used (most-used-words-buffer-1 n t))
         (word-counts (cl-first most-used))
         (total-count (float (cl-second most-used))))
    (cl-loop with data-list = '()
       with data-item = nil
       for (word count) in word-counts
       do
         (setf data-item (most-used-words--form-data word count (* 100 (/ count total-count))))
         (push data-item data-list)
       finally (return (nreverse data-list)))))

Reply:
Updated! Thanks! 😄